### ISSUE TITLE
Use shared projects

### ISSUE THREAD

**Issue by fubar-coder** [rdavisau/sockets-for-pcl#30]
Hi,

the following projects are ideal candidates for shared projects which would simplify handling with this shared code:

* Sockets.Implementation.NET
* Sockets.Implementation.WinRT

Shared projects are supported natively by VS 2015. When you already use VS 2015, I can create a separate pull request later which converts those projects to shared projects.

**Comment by rdavisau** [comment]
Yeah, I have thought on this a few times in the past. I'd hope it might improve VS performance and obviously it would reduce the chance of any errors/mistakes when adding new files to either implementation (as you have to link them into each project). 

I am on VS2015 now; I realise it is a bit of effort to consolidate but if you are keen to take it on at some stage I'd be happy to accept the PR. Thanks for your help!
